<?xml version="1.0"?>
<launch>
    <!-- Launch file arguments -->
    <arg name="map"             default="floor2"/>
    <arg name="is_sim"          default="true"/>
    <arg name="use_foxglove"    default="true"/>
    <arg name="initial_pose_x"  default="-2.65488696"/>
    <arg name="initial_pose_y"  default="-1.64422277"/>
    <arg name="initial_pose_a"  default="1.57" /> <!-- wrt to map-->
    <arg name="map_path" default="$(find-pkg-share svea_core)/maps/$(var map).yaml"/>
    <arg name="state1" default="[-7.4, -15.3, 0.9, 0.0]" />
    <arg name="state2" default="[-4.8, -11.4, 1.03, 0.0]" />    
    <arg name="points" default="['[-2.3, -7.1]', '[3.51, 1.95]', '[10.5, 11.7]', '[5.7, 15.0]', '[-1.18, 4.69]', '[-7.0, -4.0]']" />
    <arg name="vehicle1_name" default="vehicle1"/>
    <arg name="vehicle2_name" default="vehicle2"/> 

    <!-- Start Map Server and Foxglove -->
    <include file="$(find-pkg-share svea_core)/launch/map_and_foxglove.xml">
        <arg name="map" value="$(var map)" />
        <arg name="use_foxglove" value="$(var use_foxglove)" />
    </include>

    <!-- Start Vehicle 1 -->
    <group>
        <push_ros_namespace namespace="$(var vehicle1_name)" />

        <include file="$(find-pkg-share svea_core)/launch/svea_launch.xml">
            <arg name="is_sim" value="$(var is_sim)" />
            <arg name="initial_pose_x" value="$(var initial_pose_x)" />
            <arg name="initial_pose_y" value="$(var initial_pose_y)" />
            <arg name="initial_pose_a" value="$(var initial_pose_a)" />
            <arg name="state" value="$(var state1)" />
            <arg name="points" value="$(var points)" />
            <arg name="vehicle_name" value="$(var vehicle1_name)" />
        </include>   

        <!-- Start pure_pursuit -->
        <node name="pure_pursuit" pkg="svea_examples" exec="pure_pursuit.py" namespace="" output="screen">
            <param name="use_rviz" value="$(var use_foxglove)"/>
            <param name="is_sim" value="$(var is_sim)"/>
            <param name="state" value="$(var state1)" />
            <param name="points" value="$(var points)" />
        </node>
    </group>

    <!-- Start Vehicle 2 -->
    <group>
        <push_ros_namespace namespace="$(var vehicle2_name)" />

        <include file="$(find-pkg-share svea_core)/launch/svea_launch.xml">
            <arg name="is_sim" value="$(var is_sim)" />
            <arg name="initial_pose_x" value="$(var initial_pose_x)" />
            <arg name="initial_pose_y" value="$(var initial_pose_y)" />
            <arg name="initial_pose_a" value="$(var initial_pose_a)" />
            <arg name="state" value="$(var state2)" />
            <arg name="points" value="$(var points)" />
            <arg name="vehicle_name" value="$(var vehicle2_name)" />
        </include>   

        <!-- Start pure_pursuit -->
        <node name="pure_pursuit" pkg="svea_examples" exec="pure_pursuit.py" namespace="" output="screen">
            <param name="use_rviz" value="$(var use_foxglove)"/>
            <param name="is_sim" value="$(var is_sim)"/>
            <param name="state" value="$(var state2)" />
            <param name="points" value="$(var points)" />
        </node>
    </group>

</launch>
